hathor/merged_mining/digibyte_rpc.py

```diff
@@ -0,0 +1,151 @@
+from abc import ABC, abstractmethod
+from itertools import count
+from typing import Any, Callable, Dict, Iterator, List, Optional, Union, cast
+
+from aiohttp import BasicAuth, ClientSession
+from structlog import get_logger
+
+logger = get_logger()
+
+
+class RPCFailure(Exception):
+    def __init__(self, message: str, code: Optional[int] = None):
+        super().__init__(message)
+        self.code = code
+
+
+class IDigibyteRPC(ABC):
+    @abstractmethod
+    async def get_block_template(self, *, rules: List[str] = ['segwit'], algo: str = 'sha256d',
+                                 longpoll_id: Optional[str],
+                                 capabilities: List[str] = ['coinbasetxn', 'workid', 'longpoll']) -> Dict:
+        """ Method for the [GetBlockTemplate call](https://developer.bitcoin.org/reference/rpc/getblocktemplate.html).
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    async def verify_block_proposal(self, *, block: bytes) -> Optional[str]:
+        """ Method for the [GetBlockTemplate call](https://developer.bitcoin.org/reference/rpc/getblocktemplate.html).
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    async def submit_block(self, block: bytes) -> Optional[str]:
+        """ Method for the [SubmitBlock call](https://developer.bitcoin.org/reference/rpc/submitblock.html).
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    async def validate_address(self, address: str) -> Dict:
+        """ Method for the [ValidateAddress call](https://developer.bitcoin.org/reference/rpc/validateaddress.html).
+        """
+        raise NotImplementedError
+
+
+class DigibyteRPC(IDigibyteRPC):
+    """ Class for making calls to Digibyte's RPC.
+
+    References:
+
+    - https://bitcoin.org/en/developer-reference#remote-procedure-calls-rpcs
+    """
+
+    USER_AGENT = 'hathor-merged-mining'
```

Comment:
I think this could go in the conf file? Or maybe a constants file inside the merged_mining module. Just we don't duplicate it in all coins we have

Reply:
I'll move this to a `jsonrpc` module (I'm settling on that now). But I don't think this should normally be customizable. Also I want to move this away from `hathor-core` so that's one more reason not to create a conf that will be removed.

Reply:
Done.